packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -365,6 +364,7 @@ export const coreDefinitions: {[coreSettingName: string]: SettingsDefinition} =
     description: `Overrides for log levels`,
     type: SettingsType.SHAPE,
     isArray: true,
+    concatenateValues: true,
```

Comment:
I wonder if it should be the default 🤔  probably too much of a breaking change outside a major, but for the 3.x it might be a good idea to formalize how things are merged, and support a reset like:

```yaml
unsafeHttpWhitelist:
  - reset: true
  - "localhost:8080"
```

Reply:
> I wonder if it should be the default 

I'm not sure if a default value of `true` would be useful to be honest.

Looking at the arrays currently in the configuration, none of those would benefit from merging:

- `pnpIgnorePatterns`, `immutablePatterns`, `unsafeHttpWhitelist`: including user configuration here could lead to "it works on my machine" problems, which is the exact type of problem that pushed me and my colleagues away from npm
- `changesetBaseRefs`: this could genuinely be useful, but not for this specific property.
  This field actually contains two things: the name of the remote and the name of the branch. As a user I name my remotes (for work repos I use `blessed` instead of `upstream` for instance) while the project controls the branch names.
- `changesetIgnorePatterns`: it's not up to users to define what patterns to ignore when finding the changed files.

Having mergeability be opt-in requires the plugin author / yarn developer to have considered whether they actually want to merge the array, while this becomes an after-thought if it's opt-out.

> for the 3.x it might be a good idea to formalize how things are merged

Definitely! And not just for arrays but maps as well.